### PR TITLE
Remove unnecessary product tests

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java:34]: The advantage of this over a unit test was

- ensuring the td functions are part of the production bundle.
  - for plugins, we manage to violate this many times over, so it's a lesson learned
- ensuring the td functions work without `trino-main` on the classpath
  - for functions, we managed to make them work in unit tests only. I remember a case where one of the maintainers fixed plugin-provided aggregate functions 3 times, because we lacked product test coverage

I think we should keep at least one product tests' test case

**Review by findepi — COMMENTED** [review]
"Move Teradata function tests to unit tests"

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/functions/operators/TestComparison.java
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/functions/operators/TestComparison.java:62]: We definitely should retain some smoke test coverage of the engine as part of the product tests, to ensure Trino works in its final classpath/classloaders composition.

I don't believe such a detailed test like this was really meant for that, so I am fine with the removal.
It would be good to known what are the tests we want to retain. Maybe we don't have them yet?

**Review by findepi — COMMENTED** [review]
"Remove unnecessary product tests" LGTM

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java
Comment (martint) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java:34]: What setups do these tests run with? Are they executed with all the combinations of connectors and environments?

Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java:34]: i hope not! Once should be enough (and i guess this was the case)

**Review by electrum — APPROVED** [review]

**Comment by mosabua** [comment]
Consider this a reminder @martint  ... I assume you will merge this after the 436 release.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java
Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/teradata/TestTeradataFunctions.java:34]: based on the past experience i think we should have at least one test case showing that plugin-provided functions work when loaded in a real server, with real plugin classloader
https://github.com/trinodb/trino/issues/14486 (https://github.com/trinodb/trino/pull/14488)
https://github.com/trinodb/trino/issues/10341 (https://github.com/trinodb/trino/pull/10346, https://github.com/trinodb/trino/pull/10408)

**Review by findepi — DISMISSED** [review]

**Comment by mosabua** [comment]
I assume this is still valid and you will proceed with @martint

**Pull request closed without merging**
